pkg/controller/controller.go

```diff
@@ -170,6 +173,9 @@ func New(config Config) (*Controller, error) {
 	if config.Name == "" {
 		return nil, microerror.Maskf(invalidConfigError, "%T.Name must not be empty", config)
 	}
+	if config.RemovedFinalizersCacheTTL == 0 {
+		config.RemovedFinalizersCacheTTL = config.ResyncPeriod * 3
```

Comment:
If the TTL is not set we keep the current value.

Reply:
Could we put this into a variable? Same as with `DefaultResyncPeriod`?

This additionally changes behaviour slightly as this does always calculate the TTL depending on the ResyncPeriod before it is defaulted.
If the ResyncPeriod is 0 (not set, will be defaulted in Line 180) this will cause the TTL to also be 0.

Reply:
> This additionally changes behaviour slightly as this does always calculate the TTL depending on the ResyncPeriod before it is defaulted.

This is good. It was intended to be 3 times ResyncPeriod.

Reply:
Strongly disagree - this introduces IMO unintended behaviour of defaulting the TTL to 0.

Reply:
Thanks @MarcelMue that's a good catch. This conflicts with the defaulting of the resync period and needs to be changed if we take this approach.

@kopiczko Thanks for the suggestion of reducing the resync period. I think that may work better and means we don't need this.